miner/block.go

```diff
@@ -255,6 +259,12 @@ loop:
 			txs.Pop()
 			continue
 		}
+		// Same short-circuit of the gas above, but for bytes in the block (b.bytesBlock != nil => GingerbreadP2)
```

Comment:
Above there is a check if there's enough gas for another simple transaction:
```go
if b.gasPool.Gas() < params.TxGas {
	log.Trace("Not enough gas for further transactions", "have", b.gasPool, "want", params.TxGas)
	break
}
```

I think we should add something similar for the remaining bytes, so we short-circuit in case there's to little block space left to add another transaction.

Reply:
The question here is what is the smaller rlp encode possible, and is that going to change in the future? for example removing the gatewayFee and currencyFee from the legacy tx. I preferred to avoid that now, and have it as a possible optimization later

Reply:
> The question here is what is the smaller rlp encode possible

It is the native transaction (with no calldata), right? That's the lower bound and should be quite simple to calculate.